### ISSUE TITLE
Remove `pub` from various fields in exported types, notably in `extrinsic::extra`

### ISSUE THREAD

**Issue by jsdw** [paritytech/subxt#390]
Various types, notably in `crate::extrinsic::extra`, expose their inner workings (including `PhantomData`) to the world, eg:

```
pub struct ChargeAssetTxPayment<T: Config> {
    /// The tip for the block author.
    #[codec(compact)]
    pub tip: u128,
    /// The asset with which to pay the tip.
    pub asset_id: Option<u32>,
    /// Marker for unused type parameter.
    pub marker: PhantomData<T>,
}
```

and

```
pub struct ChargeTransactionPayment<T: Config>(
    #[codec(compact)] u128,
    pub PhantomData<T>,
);
```

And others. We should have a pass over the types that we export, and in cases like the above we should provide a better interface for creating/accessing fields that hides details like `PhantomData`.

As it stands, I'm not sure that most of the types in `extra` are ever even instantiated and aren't just used at the type level, but I'm not entirely clear at the mo?

**Comment by jsdw** [comment]
This is no longer relevant after #490, which reworks this stuff.
